lesson18/src/js/main.js

```diff
@@ -0,0 +1,275 @@
+    sliderImage.height = image.height;
+
+    i === 0 && sliderItem.classList.add("is-active");
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  const fragment = document.createDocumentFragment();
+  const buttonArea = createElementWithClassName("div", "slider__button");
+
+  const direction = ["previous", "next"];
+  direction.forEach((element) => {
+    const button = document.createElement("button");
+    button.classList.add("arrow", `${element}`);
+    button.id = `js-button_${element}`;
+    button.textContent = `${element}`;
+    button.setAttribute("aria-label", `${element}`);
+    element === 'previous' && (button.disabled = 'true');
+    fragment.appendChild(button);
+  });
+
+  const slideArea = document.querySelector(".mainvisual__images");
+  slideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = createElementWithClassName("div", "slider__pagination");
+  const fragment = document.createDocumentFragment();
+  const current = document.createElement("span");
+  current.id = "js-current";
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = createElementWithClassName("span", "separation");
+  separation.textContent = "/";
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.id = "js-total";
+  total.textContent = item;
+  fragment.appendChild(total);
+
+  const textArea = document.querySelector(".mainvisual__textarea");
+  textArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function renderPagenationBullet(item){
+  const pagenationArea = createElementWithClassName("div","mainvisual__pagenation__bullet");
+  const fragment = document.createDocumentFragment();
+  for(let i = 0; i < item; i++){
+    const pagenationBullet = createElementWithClassName("span", "bar");
+    pagenationBullet.style.width = "calc(width / item)";
+    pagenationBullet.setAttribute ("role", "button");
+    pagenationBullet.dataset.pagenationIndex = i + 1;
+    fragment.appendChild(pagenationBullet);
+
+    i === 0 && pagenationBullet.setAttribute("aria-current","true");
+
+  }
+  const slideArea = document.querySelector('.mainvisual__images__inner');
+  slideArea.appendChild(pagenationArea).appendChild(fragment);
+}
+
+function addNoimage(){
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const sliderImage = createElementWithClassName("img", "slider__image");
+  sliderImage.src = './images/slide-image01.png'
+  sliderImage.width = 800;
+  sliderImage.height = 800;
+  sliderImage.style.opacity = 1;
+  slideImageArea.appendChild(sliderImage);
+}
+
+function switchPagination(currentIndex) {
+  const pagenationCurrent = document.getElementById("js-current");
+  pagenationCurrent.textContent = currentIndex;
+
+}
+
+function switchImg(target) {
```

Comment:
IMO
```suggestion
function changeActive(target) {
```

Reply:
Thank you for your review.
I changed the name of the function that used switch to change.
In this review I learned that switch has the meaning of swapping.

[bed25e3](https://github.com/nyattsu72/hanson-frontend-cource/pull/26/commits/bed25e385bf2e9e48ecd77fa8984fa355209cfa7)